### PR TITLE
add chat rate limit

### PR DESCRIPTION
**Pull request by qq578023708**
allow set player chat rate limit

### CODE REVIEW

File: src/game/server/player.h [hunk 1]
```diff
@@ -622,6 +628,7 @@ class CBasePlayer : public CBaseCombatCharacter
 	// say/sayteam allowed?
 	virtual bool		CanHearAndReadChatFrom( CBasePlayer *pPlayer ) { return true; }
 	virtual bool		CanSpeak( void ) { return true; }
+	virtual PlayerSpeakResult        CheckSpeak( void );
```
Comment (BenLubar) [inline src/game/server/player.h:631]: Unless we're planning on making this a lot more complicated very soon, I'd prefer that this just returns a bool.

File: src/game/shared/swarm/asw_gamerules.cpp [hunk 2]
```diff
@@ -204,6 +204,7 @@ extern ConVar old_radius_damage;
 	ConVar rd_mapcycle_endless( "rd_mapcycle_endless", "0", FCVAR_ARCHIVE, "Automatically select the next Endless mission." );
 	ConVar rd_mapcycle_bonus( "rd_mapcycle_bonus", "1", FCVAR_ARCHIVE, "Automatically select the next Bonus mission." );
 	ConVar rd_mapcycle_ignore( "rd_mapcycle_ignore", "", FCVAR_ARCHIVE, "Comma-separated list of map filenames (no .bsp) that cannot be selected by map cycle." );
+	ConVar rd_chat_ratelimit ( "rd_chat_ratelimit", "1", FCVAR_NONE, "Set player chat rate,default 1 second" );
```
Comment (BenLubar) [inline src/game/shared/swarm/asw_gamerules.cpp:207]: Don't put the default value in the description, the engine will automatically do that so it'll end up there twice.

**Comment by jhh8** [comment]
default 0? there already is some sort of limiting, not sure if clientside or serverside

File: src/game/server/player.h (hunk 1 above)
Comment (jhh8) [reply src/game/server/player.h:631]: should just edit the canspeak function instead of a new one

**Pull request closed without merging**